### PR TITLE
feat: add Firestore and datastore multiple inequality samples

### PR DESCRIPTION
**Pull request by daniel-sanche**
touches the following region tags:

- firestore_query_filter_compound_multi_ineq
- firestore_query_indexing_considerations
- firestore_query_order_fields
- datastore_query_filter_compound_multi_ineq
- datastore_query_indexing_considerations
- datastore_query_order_fields
- firestore_query_filter_or
- firestore_query_filter_or_compound
- firestore_query_filter_not_eq
- firestore_query_filter_not_in

context: go/firestore_multiineq_sample_tracker

### CODE REVIEW

**Review by triplequark — APPROVED** [review]

File: firestore/cloud-client/query_filter_or.py [hunk 1]
```diff
@@ -12,28 +12,51 @@
...
 
-# [START firestore_query_filter_or]
 from google.cloud import firestore
-from google.cloud.firestore_v1.base_query import FieldFilter, Or
 
 
-def query_or_composite_filter(project_id: str) -> None:
+def query_or_filter(project_id: str) -> None:
     # Instantiate the Firestore client
     client = firestore.Client(project=project_id)
```
Comment (davidcavazos) [inline firestore/cloud-client/query_filter_or.py:20]: To make the sample copy-paste runnable as per the [style guide](https://googlecloudplatform.github.io/samples-style-guide/), the client creation should be included in the code sample.

Same with other samples.

File: firestore/cloud-client/query_filter_or.py [hunk 2]
```diff
@@ -12,28 +12,51 @@
...
+    query = col_ref.where(
+        Or(
+            FieldFilter("capital", "==", True),
+            FieldFilter("population", ">", 1_000_000),
+        )
+    )
+    docs = query.stream()
+    # [END firestore_query_filter_or]
 
     print("Documents found:")
```
Comment (davidcavazos) [inline firestore/cloud-client/query_filter_or.py:35]: Is it intentional that the prints were left out from the sample? Someone running the sample wouldn't see anything in their screen.

Same with other samples.

File: firestore/cloud-client/query_filter_or_test.py [hunk 3]
```diff
@@ -55,15 +59,40 @@ def delete_document_collection(data, collection):
...
+    collection = client.collection("cities")
 
     try:
         create_document_collection(data, collection)
-        query_or_composite_filter(PROJECT_ID)
+        query_or_filter(PROJECT_ID)
     finally:
         delete_document_collection(data, collection)
 
     out, _ = capsys.readouterr()
```
Comment (davidcavazos) [inline firestore/cloud-client/query_filter_or_test.py:72]: Testing on stdout is brittle and can cause tests to fail if the underlying implementation changes. Instead, you should return the response from the code snippet and test it directly here.

File: firestore/cloud-client/snippets.py [hunk 4]
```diff
@@ -990,3 +1016,42 @@ def regional_endpoint():
         print(r)
     # [END firestore_regional_endpoint]
     return cities_query
+
+
+def query_filter_compound_multi_ineq():
+    from google.cloud import firestore
+
+    db = firestore.Client()
```
Comment (davidcavazos) [inline firestore/cloud-client/snippets.py:1024]: Can we make the import and client creation as part of the sample to make it copy-paste-runnable?

Same with other samples.

File: datastore/cloud-client/query_multi_ineq.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+See https://cloud.google.com/python/docs/reference/datastore/latest before running code.
+"""
+
+
+def query_filter_compound_multi_ineq():
+    from google.cloud import datastore
+    from google.cloud.datastore.query import PropertyFilter
+
+    client = datastore.Client()
+    # [START datastore_query_filter_compound_multi_ineq]
```
[the remark this answers is not in the document]
Comment (daniel-sanche) [reply datastore/cloud-client/query_multi_ineq.py:27]: Firestore doesn't currently do this for any of their samples, across languages. 

For this PR, I'm hoping to focus on just getting the missing samples from being the hardcoded docs into source control, and then hopefully iterate on other docs improvements in the future

File: firestore/cloud-client/query_filter_or.py (hunk 1 above)
Comment (daniel-sanche) [reply firestore/cloud-client/query_filter_or.py:20]: Yeah, this is in line with the other languages: https://firebase.google.com/docs/firestore/query-data/queries#or_queries

File: firestore/cloud-client/query_filter_or.py (hunk 2 above)
Comment (daniel-sanche) [reply firestore/cloud-client/query_filter_or.py:35]: Yeah, this is in line with the other languages: https://firebase.google.com/docs/firestore/query-data/queries#or_queries. They build a query, but don't print anything from it

File: firestore/cloud-client/query_filter_or_test.py (hunk 3 above)
Comment (daniel-sanche) [reply firestore/cloud-client/query_filter_or_test.py:72]: I'm hoping to avoid re-writing the existing samples and tests for this PR

**Pull request merged**